src/lib/radio/radio.ts

```diff
@@ -117,6 +120,15 @@ export class MdRadioGroup implements AfterContentInit, ControlValueAccessor {
   }
 
   @Input()
+  get required(): boolean {
+    return this._required;
+  }
+
+  set required(value) {
+    this._required = (value != null && value !== false) ? true : null;
```

Comment:
`[attr.required]` is not the same as `[required]`. `[attr.required]` set the attribute `required` in the html element, and when the value is null the attribute `required` is removed. 

The `[required]` calls the `set required` in the component.